modules/screenshots/assets/js/preview/screenshot.js

```diff
@@ -0,0 +1,352 @@
+			);
+
+			this.createFakeContent();
+		}
+
+		this.hideUnnecessaryElements();
+		this.removeUnnecessaryElements();
+		this.handleIFrames();
+		this.loadExternalCss();
+		this.loadExternalImages();
+
+		return Promise.resolve()
+			.then( this.createImage.bind( this ) )
+			.then( this.createImageElement.bind( this ) )
+			.then( this.cropCanvas.bind( this ) )
+			.then( this.save.bind( this ) )
+			.then( this.screenshotSucceed.bind( this ) )
+			.catch( this.screenshotFailed.bind( this ) );
+	}
+
+	/**
+	 * Fake content for documents that dont have any content.
+	 */
+	createFakeContent() {
+		this.elements.$elementor = jQuery( '<div></div>' ).css( {
+			height: this.getSettings( 'crop.height' ),
+			width: this.getSettings( 'crop.width' ),
+			display: 'flex',
+			alignItems: 'center',
+			justifyContent: 'center',
+		} );
+
+		this.elements.$elementor.append(
+			jQuery( '<h1></h1>' ).css( { fontSize: '85px' } ).html( this.getSettings( 'empty_content_headline' ) )
+		);
+
+		document.body.prepend( this.elements.$elementor );
+	}
+
+	/**
+	 * CSS from another server cannot be loaded with the current dom to image library.
+	 * this method take all the links from another domain and proxy them.
+	 */
+	loadExternalCss() {
+		const excludedUrls = [
+			this.getSettings( 'home_url' ),
+			...this.getSettings( 'excluded_external_css_urls' ),
+		];
+
+		const notSelector = excludedUrls
+			.map( ( url ) => `[href^="${ url }"]` )
+			.join( ', ' );
+
+		jQuery( 'link' ).not( notSelector ).each( ( index, el ) => {
+			const $link = jQuery( el ),
+				$newLink = $link.clone();
+
+			$newLink.attr( 'href', this.getScreenshotProxyUrl( $link.attr( 'href' ) ) );
+
+			jQuery( 'head' ).append( $newLink );
+			$link.remove();
+		} );
+	}
+
+	/**
+	 * Make a proxy to images urls that has some problems with cross origin (like youtube).
+	 */
+	loadExternalImages() {
+		const selector = this.getSettings( 'external_images_urls' )
+			.map( ( url ) => `img[src^="${ url }"]` )
+			.join( ', ' );
+
+		jQuery( selector ).each( ( index, el ) => {
+			const $img = jQuery( el );
+
+			$img.attr( 'src', this.getScreenshotProxyUrl( $img.attr( 'src' ) ) );
+		} );
+	}
+
+	/**
+	 * Html to images libraries can not snapshot IFrames
+	 * this method convert all the IFrames to some other elements.
+	 */
+	handleIFrames() {
+		this.elements.$elementor.find( 'iframe' ).each( ( index, el ) => {
+			const $iframe = jQuery( el ),
+				$iframeMask = jQuery( '<div />', {
+					css: {
+						background: 'gray',
+						width: $iframe.width(),
+						height: $iframe.height(),
+					},
+				} );
+
+			$iframe.before( $iframeMask );
+			$iframe.remove();
+		} );
+	}
+
+	/**
+	 * Hide all the element except for the target element.
+	 */
+	hideUnnecessaryElements() {
+		jQuery( 'body' ).prepend(
+			this.elements.$elementor
+		);
+
+		jQuery( 'body > *' ).not( this.elements.$elementor ).css( 'display', 'none' );
```

Comment:
Consider caching `jQuery( 'body > *' ).not( this.elements.$elementor )` inside `getDefaultElements`